gio/src/action_entry.rs

```diff
@@ -69,13 +69,13 @@ where
         })
     }
 
-    pub fn parameter_type(mut self, parameter_type: &str) -> Self {
-        self.0.parameter_type = Some(parameter_type.to_owned());
+    pub fn parameter_type(mut self, parameter_type: Option<&VariantTy>) -> Self {
+        self.0.parameter_type = parameter_type.map(|vt| vt.to_owned());
         self
     }
 
-    pub fn state(mut self, state: &str) -> Self {
-        self.0.state = Some(state.to_owned());
+    pub fn state(mut self, state: Variant) -> Self {
```

Comment:
Here I decided to take the variant by value... this avoids a copy since I assume in the common case this will be some state created on the spot (eg `false.to_variant()`). However `SimpleAction::new_stateful` takes it by reference, let me know if you prefer to change it for consistency

Reply:
Variants are refcounted so this is not too important, but seems nicer like this indeed. Can you change the other one too?